palak/src/components/withLoading.jsx

```diff
@@ -0,0 +1,29 @@
+// import React, { Children } from "react";
+
+// const withLoading = (WrappedComponent) => {
+//   return ({ ...props }) => {
+//     return <WrappedComponent {...props}></WrappedComponent>;
+//   };
+// };
+import { ImSpinner3 } from "react-icons/im";
+
+function withLoading(WrappedComponent) {
+  return ({ isLoading, children, ...props }) => (
+    <WrappedComponent {...props}>
+      <div>
```

Comment:
Check if this div is required or not as it is seeming if there's div inside div without parent classnames.